### PR TITLE
Bug fixes, unit tests and an updated CertLogicEngine (EXPOSUREAPP-8060)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Bug fixes, unit tests and an updated external library

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/CertLogicEngineWrapperTest.kt [hunk 1]
```diff
@@ -62,4 +89,79 @@ class CertLogicEngineWrapperTest : BaseTest() {
...
+        val container = valueSets.toValueSetsContainer(languageCode = Locale.GERMAN)
+        coEvery { valueSetsRepository.latestVaccinationValueSets } returns
+            flowOf(container.vaccinationValueSets)
+        coEvery { valueSetsRepository.latestTestCertificateValueSets } returns
+            flowOf(container.testCertificateValueSets)
+        valueSetWrapper = ValueSetWrapper(valueSetsRepository, dccValidationRepository)
+        wrapper = CertLogicEngineWrapper(valueSetWrapper)
+
+        json.testCases.forEachIndexed { index, certLogicTestCase ->
+            if (certLogicTestCase.description == "EP: issuerCountryCode shall be retrieved CWT (sample TC)")
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/CertLogicEngineWrapperTest.kt:114]: needs to be removed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt [hunk 2]
```diff
@@ -124,21 +126,29 @@ internal fun String.toZonedDateTime(): ZonedDateTime {
 internal val ZonedDateTime.asExternalString: String
     get() = this.toString()
 
-internal val DccData<out DccV1.MetaData>.type: String
+internal val DccData<out DccV1.MetaData>.asExternalType: CertificateType
+    get() = when (certificate) {
+        is VaccinationDccV1 -> CertificateType.VACCINATION
+        is TestDccV1 -> CertificateType.TEST
+        is RecoveryDccV1 -> CertificateType.RECOVERY
+        else -> throw IllegalArgumentException("Unknown certificate type.")
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt:134]: should we treat other types as `CertificateType.GENERAL` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/CertLogicEngineWrapper.kt [hunk 3]
```diff
@@ -16,19 +19,25 @@ import javax.inject.Inject
...
+    private val valueSetWrapper: ValueSetWrapper,
+    private val dccJsonSchema: DccJsonSchema,
 ) {
 
     private val engine: DefaultCertLogicEngine by lazy {
-        DefaultCertLogicEngine(DefaultJsonLogicValidator())
+        DefaultCertLogicEngine(
+            DefaultAffectedFieldsDataRetriever(
+                schemaJsonNode = ObjectMapper().readTree(dccJsonSchema.rawSchema),
+                objectMapper = ObjectMapper()
```
[the remark this answers is not in the document]
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/CertLogicEngineWrapper.kt:30]: If possible use the injected one:
`@BaseJackson private val objectMapper: ObjectMapper`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt:134]: There is no CertificateType.GENERAL any more.

**Review by mtwalli — DISMISSED** [review]
Lgtm

**Review by mtwalli — APPROVED** [review]

**Review by axelherbstreith — APPROVED** [review]

**Pull request merged**